### ISSUE TITLE
Clear text cred display in mirror settings

### ISSUE THREAD

**Issue by handnot2** [go-gitea/gitea#672]
- Gitea version (or commit ref): 1.0.1
- Git version:
- Operating system: Debian Jesse arm
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

Create a mirror of a private repo by providing the GIT URL and credentials at the top of the
`Mirror Settings` section. Once the mirror is created, the page
(`Settings` > `Mirror Settings` > `Mirror Address`) shows the credential in clear (https://`user:passwd`@host:port/repo..).

It is better to remove this mirror address display. If it is still needed, at the least mask the password.

**Comment by bkcsoft** [comment]
Just using `net/url` doing the following should suffice (Pseudo-code, variables might be off, here be dragons)
```go
// scheme://[userinfo@]host/path[?query][#fragment]
url, err := url.Parse(repo.MirrorAddress)
// err-check
url.User = nil
ctx.Data["MirrorAddress"] = url.String()
```
Ref: https://golang.org/pkg/net/url/#URL.String
> if u.User is nil, userinfo@ is omitted.

**Comment by bkcsoft** [comment]
You can also (if you want to see that auth is used) do the following to show `https://******@example.com/foobar.git`
```go
if url.User {
  url.User = url.User(`********`)
}
```
Ref: https://golang.org/pkg/net/url/#User

**Comment by tboerger** [comment]
IMHO this is not a real security issue because it's anyway only visible to repo admins. Beside that just replacing or wiping credentials without any workaround makes it impossible to update the settings without retyping the credentials over and over again.

**Comment by bkcsoft** [comment]
@tboerger to me it feels like a "over-the-shoulder" security issue, which IMO should be fixed. Though showing and editing doesn't have to be the same "box"

**Comment by mrsdizzie** [comment]
Fixed in #7651
